src/vector/app.tsx

```diff
@@ -197,48 +197,58 @@ async function verifyServerConfig() {
         // care if they are syntactically correct though, so we shove them through the .well-known
         // validators for that purpose.
 
+        let discoveryResult = null;
+
         const config = SdkConfig.get();
         let wkConfig = config['default_server_config']; // overwritten later under some conditions
-        const serverName = config['default_server_name'];
-        const hsUrl = config['default_hs_url'];
-        const isUrl = config['default_is_url'];
-
-        const incompatibleOptions = [wkConfig, serverName, hsUrl].filter(i => !!i);
-        if (incompatibleOptions.length > 1) {
-            // noinspection ExceptionCaughtLocallyJS
-            throw newTranslatableError(_td(
-                "Invalid configuration: can only specify one of default_server_config, default_server_name, " +
-                "or default_hs_url.",
-            ));
-        }
-        if (incompatibleOptions.length < 1) {
-            // noinspection ExceptionCaughtLocallyJS
-            throw newTranslatableError(_td("Invalid configuration: no default server specified."));
-        }
 
-        if (hsUrl) {
-            console.log("Config uses a default_hs_url - constructing a default_server_config using this information");
-            console.warn(
-                "DEPRECATED CONFIG OPTION: In the future, default_hs_url will not be accepted. Please use " +
-                "default_server_config instead.",
-            );
+        let serverName = null;
+        if (fragParams.serverName) {
```

Comment:
This should probably respect the `config['disable_custom_urls']` option when it's set to `true`.

However I didn't tested it at all, so not really sure, maybe it's handled on matrix-react-sdk side. :man_shrugging: 
```suggestion
        if (fragParams.serverName && !config['disable_custom_urls']) {
```